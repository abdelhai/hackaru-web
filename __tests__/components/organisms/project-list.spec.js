import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import ProjectList from '@/components/organisms/project-list';
import ActivityEditor from '@/components/organisms/activity-editor';
import ProjectEditor from '@/components/organisms/project-editor';

describe('ProjectList', () => {
  let factory;
  let wrapper;

  const $store = new Store({
    getters: {
      'projects/getProjects': [
        {
          id: 1,
          name: 'Development',
          color: '#ff0'
        },
        {
          id: 2,
          name: 'Review',
          color: '#f00'
        }
      ]
    }
  });

  beforeEach(() => {
    $store.reset();
    factory = new Factory(ProjectList, { mocks: { $store } });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when click left arrow button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.left-arrow-button').vm.$emit('click');
    });

    it('emit pop', () => {
      expect(wrapper.emitted('pop')[0]).toEqual([
        { component: ActivityEditor }
      ]);
    });
  });

  describe('when click add button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.add-button').vm.$emit('click');
    });

    it('emit push', () => {
      expect(wrapper.emitted('push')[0]).toEqual([
        { component: ProjectEditor }
      ]);
    });
  });

  describe('when click edit-button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper
        .findAll('.edit-button')
        .at(1)
        .vm.$emit('click');
    });

    it('emit push', () => {
      expect(wrapper.emitted('push')[0]).toEqual([
        {
          component: ProjectEditor,
          params: {
            id: 2,
            name: 'Review',
            color: '#f00'
          }
        }
      ]);
    });
  });

  describe('when click project-content', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper
        .findAll('.project-content')
        .at(2)
        .trigger('click');
    });

    it('emit pop', () => {
      expect(wrapper.emitted('pop')[0]).toEqual([
        {
          component: ActivityEditor,
          params: {
            project: {
              id: 2,
              name: 'Review',
              color: '#f00'
            }
          }
        }
      ]);
    });
  });
});
