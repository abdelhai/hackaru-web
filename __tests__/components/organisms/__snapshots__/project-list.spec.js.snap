// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList render correctly 1`] = `
<section>
  <modal-header-stub>
    <btn-stub
      aria-label="ariaLabels.back"
      class="left-arrow-button has-icon"
      type="button"
    >
      <icon-stub
        class="is-large"
        name="chevron-left-icon"
      />
    </btn-stub>
     
    <h1>
      title
    </h1>
     
    <btn-stub
      aria-label="ariaLabels.add"
      class="add-button has-icon"
      type="button"
    >
      <icon-stub
        name="plus-icon"
      />
    </btn-stub>
  </modal-header-stub>
   
  <div
    class="list-item"
  >
    <div
      class="project-content"
    >
      <project-name-stub
        color="#cccfd9"
        name="No Project"
      />
       
      <icon-stub
        class="check-icon"
        name="check-icon"
      />
    </div>
     
    <!---->
  </div>
  <div
    class="list-item"
  >
    <div
      class="project-content"
    >
      <project-name-stub
        color="#ff0"
        name="Development"
      />
       
      <!---->
    </div>
     
    <btn-stub
      aria-label="ariaLabels.edit"
      class="has-icon edit-button"
      type="button"
    >
      <icon-stub
        class="is-primary"
        name="edit-3-icon"
      />
    </btn-stub>
  </div>
  <div
    class="list-item"
  >
    <div
      class="project-content"
    >
      <project-name-stub
        color="#f00"
        name="Review"
      />
       
      <!---->
    </div>
     
    <btn-stub
      aria-label="ariaLabels.edit"
      class="has-icon edit-button"
      type="button"
    >
      <icon-stub
        class="is-primary"
        name="edit-3-icon"
      />
    </btn-stub>
  </div>
</section>
`;
